templates/admin/accounts/hostingprovider/hosting_provider_fieldset.html

```diff
@@ -0,0 +1,60 @@
+<fieldset class="module aligned {{ fieldset.classes }}">
+    {% if fieldset.name %}<h2>{{ fieldset.name }}</h2>{% endif %}
+    {% if fieldset.description %}
+        <div class="description">{{ fieldset.description|safe }}</div>
+    {% endif %}
+    {% for line in fieldset %}
+        <div class="form-row{% if line.fields|length_is:'1' and line.errors %} errors{% endif %}{% if not line.has_visible_field %} hidden{% endif %}{% for field in line %}{% if field.field.name %} field-{{ field.field.name }}{% endif %}{% endfor %}">
+            {% if line.fields|length_is:'1' %}{{ line.errors }}{% endif %}
+            {% for field in line %}
+                <div{% if not line.fields|length_is:'1' %} class="fieldBox{% if field.field.name %} field-{{ field.field.name }}{% endif %}{% if not field.is_readonly and field.errors %} errors{% endif %}{% if field.field.is_hidden %} hidden{% endif %}"{% elif field.is_checkbox %} class="checkbox-row"{% endif %}>
+                    {% if not line.fields|length_is:'1' and not field.is_readonly %}{{ field.errors }}{% endif %}
+                    {% if field.is_checkbox %}
+                        {{ field.field }}{{ field.label_tag }}
+                    {% else %}
+                        {{ field.label_tag }}
+                        {% if field.is_readonly %}
+                            <div class="readonly">{{ field.contents }}</div>
+                        {% else %}
+                            {{ field.field }}
+                        {% endif %}
+                    {% endif %}
+                    {% if field.field.help_text %}
+                        <div class="help">{{ field.field.help_text|safe }}</div>
+                    {% endif %}
+                </div>
+            {% endfor %}
+        </div>
+    {% endfor %}
+
+    <h2>Authorised hosting provider users</h2>
+    {% with original.user_set.all as users %}
+
+    {% if users %}
+      <div class="form-row">
+      <p>These are the users able to update this hosting provider:</p>
+      </div>
+      {% for user in users %}
+      <div class="form-row">
+        <div class="fieldBox">
+          <a href="{% url 'admin:accounts_user_change' user.id %}">{{  user }}</a>
+        </div>
+        {% if user.date_joined %}
+        <div class="fieldBox">
+          Date joined: {{  user.date_joined }}
+        </div>
+        {% endif %}
+        <div class="fieldBox">
+          Last login: {{  user.last_login }}
+        </div>
+      </div>
+      {% endfor%}
+    {% else %}
+      <div class="form-row">
+      <p>These no users associated with hosting provider.:</p>
```

Comment:
A bit of a typo, it should probably be `There are no users associated with the hosting provider`.